src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -584,10 +569,12 @@ private struct CompilationInfo
                 public Compilation Compilation { get; }
                 public bool HasCompleteReferences { get; }
 
-                public CompilationInfo(Compilation compilation, bool hasCompleteReferences)
+                public CompilationInfo(Compilation compilation, ProjectInfo projectInfo, bool hasCompleteReferences)
                 {
                     this.Compilation = compilation;
-                    this.HasCompleteReferences = hasCompleteReferences;
+
+                    // to have complete reference, projectInfo should be complete and we could create complete reference for this compilation.
+                    this.HasCompleteReferences = projectInfo.IsComplete && hasCompleteReferences;
```

Comment:
when IsComplete is false, it means we have no idea whether we got all references or not. so, we should assume worst which is we didn't get all references.